rclpy/rclpy/node.py

```diff
@@ -747,6 +902,8 @@ def set_descriptor(
         """
         Set a new descriptor for a given parameter.
 
+        The name in the descriptor is ignored and set to :param:`name`.
```

Comment:
I know this is out of the scope of this PR, but what is the motivation of `set_descriptor`?
I don't see why someone would want to use it. Maybe, it shouldn't be allowed (we don't have an equivalent in rclcpp).

Reply:
This was part of the scope of https://github.com/ros2/rclpy/pull/345; here's the comment where it was proposed: https://github.com/ros2/rclpy/pull/325#discussion_r280534305.

I think it makes sense in some cases -perhaps not most of them- but it's not bad either. You could change the limits for a parameter without requiring to re-declare it, or you could make it read-only. On top of that, you can specify the descriptor after declaring the parameter.